In [7]:
# Proyecto de modelo de análisis de datos en Python
# Alumno: Ángel Tomás Moreno Senén
# Curso de especialización programación con Python
# Curso escolar 2025/2026
# IES Severo Ochoa

In [8]:
# El primer paso, importar pandas
import pandas as pd


In [9]:
# Subir los archivos
train_df = pd.read_csv("train.csv")
test_df = pd.read_csv("test.csv")


In [10]:
# A ver que lleva train
train_df.head()
train_df.info()
train_df.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 38250 entries, 0 to 38249
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   id                 38250 non-null  int64 
 1   ANYO_CONSTRUCCION  36430 non-null  object
 2   CODIGO_POSTAL      36255 non-null  object
 3   USO_EDIFICIO       36281 non-null  object
 4   MUNICIPIO          36295 non-null  object
 5   PROVINCIA          36248 non-null  object
 6   TIPO_REGISTRO      36275 non-null  object
 7   FICHERO_ORIGEN     36447 non-null  object
 8   CONSUMO_EP_VALOR   38250 non-null  int64 
dtypes: int64(2), object(7)
memory usage: 2.6+ MB


,id,CONSUMO_EP_VALOR
count,38250.000000,38250.000000
mean,25023.821778,174.084078
std,14420.280457,172.779020
min,1.000000,1.000000
25%,12537.250000,119.000000
50%,24978.000000,161.000000
75%,37528.750000,213.000000
max,50000.000000,22479.000000


# HASTA AQUÍ VA.......... BIEEN

In [11]:
#Verificación de valores nulos
print(train_df.isnull().sum())

id                      0
ANYO_CONSTRUCCION    1820
CODIGO_POSTAL        1995
USO_EDIFICIO         1969
MUNICIPIO            1955
PROVINCIA            2002
TIPO_REGISTRO        1975
FICHERO_ORIGEN       1803
CONSUMO_EP_VALOR        0
dtype: int64


Esto aquí y ahora da lo siguiente:
id                      0  
ANYO_CONSTRUCCION    1820  
CODIGO_POSTAL        1995  
USO_EDIFICIO         1969  
MUNICIPIO            1955  
PROVINCIA            2002  
TIPO_REGISTRO        1975  
FICHERO_ORIGEN       1803  
CONSUMO_EP_VALOR        0  
dtype: int64


LA VARIABLE A PREDECIR es CONSUMO_EP_VALOR, Y ES: valor numérico continuo y muy disperso y relacionado con el consumo energético.

El resto de columnas: variables predictoras.

In [12]:
# Porcentaje de nulos
porcentaje_nulos = (train_df.isnull().sum() / len(train_df)) * 100
print(porcentaje_nulos.sort_values(ascending=False))

PROVINCIA            5.233987
CODIGO_POSTAL        5.215686
TIPO_REGISTRO        5.163399
USO_EDIFICIO         5.147712
MUNICIPIO            5.111111
ANYO_CONSTRUCCION    4.758170
FICHERO_ORIGEN       4.713725
id                   0.000000
CONSUMO_EP_VALOR     0.000000
dtype: float64


SON UNOS NULOS ALREDEDOR DEL 5% ... continuará
